pykokkos/core/parsers/parser.py

```diff
@@ -123,19 +124,165 @@ def get_entities(self, style: PyKokkosStyles) -> Dict[str, PyKokkosEntity]:
 
         for i, node in enumerate(self.tree.body):
             if check_entity(node, self.pk_import):
-                start: int = node.lineno - 1
 
+                start: int = node.lineno - 1
                 try:
                     stop: int = self.tree.body[i + 1].lineno - 1
                 except IndexError:
                     stop = len(self.lines)
-
+                
                 name: str = node.name
+
                 entity = PyKokkosEntity(style, cppast.DeclRefExpr(name), node, (self.lines[start:stop], start), self.path, self.pk_import)
                 entities[name] = entity
 
         return entities
 
+
+    def fix_types(self, entity: PyKokkosEntity, updated_types: UpdatedTypes) -> ast.AST:
+
+        '''
+        updatedTypes: object that contains info about inferred types
+
+        This method will walk the ast for a workunit and add any mission annotations
+        This method will also invoke fix_viewlayouts that will add missing decorators for user specified layouts
+        '''
+        
+        check_entity: Callable[[ast.stmt], bool]
+        style: PyKokkosStyles = entity.style
+
+        # only supports standalone workunits, return the entity AST as it is
+        if style is not PyKokkosStyles.workunit or updated_types == None:
+            return entity.AST
+        
+        check_entity = self.is_workunit
+
+        # For now, just so we can raise an error instead of unexpectedly crashing
+        primitives_supported = ["int", "bool", "float"]
+        entity_tree: ast.AST = None
+        
+        #*2 Changing annotations for the needed workunit definitions
+        for node in self.tree.body:
+
+            # Check for the workunit we need
+            if check_entity(node, self.pk_import) and updated_types.workunit.__name__ == node.name:
+                
+                entity_tree = node
+
+                # if modifications to layout decorator is needed
+                if len(updated_types.layout_change):
+                    node.decorator_list = self.fix_viewlayout(node, updated_types.layout_change)
+
+                for arg_obj in node.args.args:
+                    if arg_obj.arg in updated_types.inferred_types:
+                        update_type = updated_types.inferred_types[arg_obj.arg]
+
+                        if update_type in primitives_supported:
+                            arg_obj.annotation = ast.Name(id=update_type, ctx=ast.Load())
+                                                        
+                        elif "View" in update_type:
+                            # update_type = View1D:double
+                            view_type, dtype = update_type.split(':')
+
+                            arg_obj.annotation = ast.Subscript(
+                                value = ast.Attribute(
+                                    value = ast.Name(id="pk", ctx=ast.Load()),
```

Comment:
Instead of "pk", you should be using `pk_import`. That way if a user has `import pykokkos as pyk' it would still work.